### PR TITLE
Source faker: change some fields to be integer

### PR DESCRIPTION
**Pull request by edgao**
was slightly confused by ids and year being floats. change them to number.

presumably not a true breaking change since all the data was already integer?

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`84ccc76f67`](https://github.com/airbytehq/airbyte/pull/29213/commits/84ccc76f67d22b02ebaabcfe1aed9c95283bc8be)) - ❌

⏲️  Total pipeline duration: 01mn21s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691508281/84ccc76f67d22b02ebaabcfe1aed9c95283bc8be/source-faker/4.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:84ccc76f67d22b02ebaabcfe1aed9c95283bc8be)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`84ccc76f67`](https://github.com/airbytehq/airbyte/pull/29213/commits/84ccc76f67d22b02ebaabcfe1aed9c95283bc8be)) - ❌

⏲️  Total pipeline duration: 01mn40s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691508456/84ccc76f67d22b02ebaabcfe1aed9c95283bc8be/source-faker/4.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:84ccc76f67d22b02ebaabcfe1aed9c95283bc8be)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`3a78858712`](https://github.com/airbytehq/airbyte/pull/29213/commits/3a788587123f041057942b6da1f02d329963e194)) - ❌

⏲️  Total pipeline duration: 01mn19s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691508682/3a788587123f041057942b6da1f02d329963e194/source-faker/4.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3a788587123f041057942b6da1f02d329963e194)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`2f6081751f`](https://github.com/airbytehq/airbyte/pull/29213/commits/2f6081751f6920bb439b0da49bc54ad294bfe309)) - ❌

⏲️  Total pipeline duration: 02mn28s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691509852/2f6081751f6920bb439b0da49bc54ad294bfe309/source-faker/4.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2f6081751f6920bb439b0da49bc54ad294bfe309)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by evantahler** [comment]
It's a breaking change ;) 

```
E       connector_acceptance_test.utils.backward_compatibility.NonBackwardCompatibleError: BackwardIncompatibilityContext.DISCOVER - The'type' field value was changed.. Diff: Value of root['products']['properties']['id']['type'] changed from "number" to "integer".
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`e30621cc1e`](https://github.com/airbytehq/airbyte/pull/29213/commits/e30621cc1ef2a1b5615beee4e93bafe36b0f53d5)) - ❌

⏲️  Total pipeline duration: 02mn45s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ❌        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ❌        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691536950/e30621cc1ef2a1b5615beee4e93bafe36b0f53d5/source-faker/5.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e30621cc1ef2a1b5615beee4e93bafe36b0f53d5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`843e6f7649`](https://github.com/airbytehq/airbyte/pull/29213/commits/843e6f7649704d4fd32c4be960d676163c7db2d4)) - ❌

⏲️  Total pipeline duration: 02mn25s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691537818/843e6f7649704d4fd32c4be960d676163c7db2d4/source-faker/5.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:843e6f7649704d4fd32c4be960d676163c7db2d4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`386045db5b`](https://github.com/airbytehq/airbyte/pull/29213/commits/386045db5b50a4b4a6326a22ce215292c1d5d55d)) - ❌

⏲️  Total pipeline duration: 02mn04s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691538593/386045db5b50a4b4a6326a22ce215292c1d5d55d/source-faker/5.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:386045db5b50a4b4a6326a22ce215292c1d5d55d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by octavia-squidington-iii** [comment]
##  source-faker test report (commit [`c1e8cf503f`](https://github.com/airbytehq/airbyte/pull/29213/commits/c1e8cf503fbc8710da43995f83f6162815599395)) - ✅

⏲️  Total pipeline duration: 02mn26s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-faker/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |
| Code format checks                                                  | ✅        |
| Connector package install                                           | ✅        |
| Build source-faker docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                          | ✅        |
| Acceptance tests                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_faker_schema/1691538991/c1e8cf503fbc8710da43995f83f6162815599395/source-faker/5.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:c1e8cf503fbc8710da43995f83f6162815599395)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

File: airbyte-integrations/connectors/source-faker/metadata.yaml [hunk 1]
```diff
@@ -32,6 +32,9 @@ data:
     - language:python
   releases:
     breakingChanges:
+      5.0.0:
+        message: "ID and products.year fields are changing to be integers instead of floats."
+        upgradeDeadline: "2023-08-31"
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-faker/metadata.yaml:37]: cc @erohmensing - a wild breaking change has appeared

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/source-faker/metadata.yaml (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/source-faker/metadata.yaml:37]: I swear I'm not _trying_ to be a conn ops early adopter, it just keeps happening :p

**Pull request merged**